pkg/kubelet/container_reference_manager.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"sync"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+)
+
+// ContainerRefManager manages the references for the containers.
+// The references are used for reporting events such as creation,
+// failure, etc.
+type ContainerRefManager struct {
+	sync.RWMutex
+	// TODO(yifan): Change type from string to types.UID.
```

Comment:
@vmarmol I think we might want to change to the type of Container.ID in pkg/kubelet/container/runtime.go later